### PR TITLE
Add channel_join script command

### PR DESCRIPTION
**Pull request by Emistry**
* **Addressed Issue(s)**: 
The existing channel system that used by NPC script, doesn't have a method that add players into an existing channel, the only method it have, is the auto join upon login.
This is inconvenient when try to create a new channel, and unable to configure who should auto join the channel by npc script

* **Server Mode**: 
any

* **Description of Pull Request**: 
enable NPC script to configure player to join a channel.

### CODE REVIEW

**Review by aleos89 — APPROVED** [review]

File: doc/script_commands.txt [hunk 1]
```diff
@@ -11132,6 +11132,13 @@ Returns 1 on success.
 
 ---------------------------------------
 
+*channel_join "<chname>"{, <char_id>};
```
Comment (Lemongrass3110) [inline doc/script_commands.txt:11135]:
```suggestion
*channel_join "<channel name>"{, <char_id>};
```
Just to make clear that this is not interpreted as character name.

File: doc/script_commands.txt [hunk 2]
```diff
@@ -11132,6 +11132,13 @@ Returns 1 on success.
...
+
+Join an existing public or private channel.
+Returns 1 on success.
```
Comment (Lemongrass3110) [inline doc/script_commands.txt:11138]:
```suggestion
Returns true on success.
```

File: src/map/script.cpp [hunk 3]
```diff
@@ -24656,6 +24656,32 @@ BUILDIN_FUNC(channel_create) {
...
+// Join an existing public or private channel.
+// Returns 1 on success.
+// ===================================
+BUILDIN_FUNC(channel_join) {
+	map_session_data *sd;
+
+	if (!script_charid2sd(3, sd))
+		return SCRIPT_CMD_FAILURE;
+
+	struct Channel *ch = NULL;
```
Comment (Lemongrass3110) [inline src/map/script.cpp:24670]:
```suggestion
	struct Channel *ch = nullptr;
```

File: src/map/script.cpp [hunk 4]
```diff
@@ -24656,6 +24656,32 @@ BUILDIN_FUNC(channel_create) {
...
 
+// ===================================
+// *channel_join "<chname>"{, <char_id>};
+// Join an existing public or private channel.
+// Returns 1 on success.
+// ===================================
+BUILDIN_FUNC(channel_join) {
+	map_session_data *sd;
+
+	if (!script_charid2sd(3, sd))
```
Comment (Lemongrass3110) [inline src/map/script.cpp:24667]:
```suggestion
	if (!script_charid2sd(3, sd)){
		script_pushint(st, false);
```

File: src/map/script.cpp [hunk 5]
```diff
@@ -24656,6 +24656,32 @@ BUILDIN_FUNC(channel_create) {
...
+		return SCRIPT_CMD_FAILURE;
```
Comment (Lemongrass3110) [inline src/map/script.cpp:24668]:
```suggestion
		return SCRIPT_CMD_FAILURE;
	}
```

File: src/map/script.cpp [hunk 6]
```diff
@@ -24656,6 +24656,32 @@ BUILDIN_FUNC(channel_create) {
...
+
+	struct Channel *ch = NULL;
+	const char *chname = script_getstr(st, 2);
+
+	if (!(ch = channel_name2channel((char *)chname, NULL, 0))) {
```
Comment (Lemongrass3110) [inline src/map/script.cpp:24673]:
```suggestion
	if (!(ch = channel_name2channel((char *)chname, nullptr, 0))) {
```

File: src/map/script.cpp [hunk 7]
```diff
@@ -24656,6 +24656,32 @@ BUILDIN_FUNC(channel_create) {
...
+		ShowError("buildin_channel_join: Channel name '%s' is invalid.\n", chname);
+		script_pushint(st, 0);
```
Comment (Lemongrass3110) [inline src/map/script.cpp:24675]:
```suggestion
		script_pushint(st, false);
```

File: src/map/script.cpp [hunk 8]
```diff
@@ -24656,6 +24656,32 @@ BUILDIN_FUNC(channel_create) {
...
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	channel_join(ch, sd);
+	script_pushint(st, 1);
```
Comment (Lemongrass3110) [inline src/map/script.cpp:24680]:
```suggestion
	script_pushint(st, true);
```

File: doc/script_commands.txt [hunk 9]
```diff
@@ -11132,6 +11132,13 @@ Returns 1 on success.
 
 ---------------------------------------
 
+*channel_join "<chname>"{, <char_id>};
+
+Join an existing public or private channel.
```
Comment (Lemongrass3110) [inline doc/script_commands.txt:11137]: I dont really see a check for `CHAN_TYPE_*`
Maybe it is not really necessary, but just wanted to point it out.

File: src/map/script.cpp [hunk 10]
```diff
@@ -24656,6 +24656,32 @@ BUILDIN_FUNC(channel_create) {
...
+	struct Channel *ch = NULL;
+	const char *chname = script_getstr(st, 2);
+
+	if (!(ch = channel_name2channel((char *)chname, NULL, 0))) {
+		ShowError("buildin_channel_join: Channel name '%s' is invalid.\n", chname);
+		script_pushint(st, 0);
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	channel_join(ch, sd);
```
Comment (Lemongrass3110) [inline src/map/script.cpp:24679]: The result of  this command should actually be checked and maybe even returned to the script engine.

**Review by Lemongrass3110 — CHANGES_REQUESTED** [review]

File: doc/script_commands.txt (hunk 9 above)
Comment (Emistry) [reply doc/script_commands.txt]: I think its not really necessary.

**Review by Lemongrass3110 — APPROVED** [review]

**Pull request merged**